pkg/upload/upload.go

```diff
@@ -0,0 +1,280 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
```

Comment:
I would leave this in `/api` package for now, and just re-use the function across the api. End then we can separate it into the pkg. Just because I am not sure should we have an upload pkg, or maybe general transfer., or file  or something like this.

Reply:
done! i will make sure to create another issue to detail these next steps.

Reply:
#462